runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -414,12 +482,14 @@ impl<Cfg: Config> Module<Cfg> {
 
         // Move the difference from the fee accumulator back into the
         // EVM token pool.
-        Cfg::Accounts::move_from_fee_accumulator(
-            ctx,
-            *ADDRESS_EVM_TOKENS,
-            &token::BaseUnits::new(return_fee.as_u128(), den),
-        )
-        .map_err(|_| Error::InsufficientBalance)?;
+        if !is_sim {
+            Cfg::Accounts::move_from_fee_accumulator(
```

Comment:
Should `move_from_fee_accumulator` (and similar) check if ctx is in simulation mode and just not do the move?